vcd/resource_vcd_nsxt_vapp_raw_test.go

```diff
@@ -139,11 +190,31 @@ resource "vcd_vapp" "{{.VappName}}" {
   depends_on   = [vcd_network_routed_v2.{{.NetworkName}}]
 }
 
-resource "vcd_vapp_org_network" "vappNetwork1" {
+resource "vcd_vapp_org_network" "routed" {
   org                = "{{.Org}}"
   vdc                = "{{.Vdc}}"
   vapp_name          = vcd_vapp.{{.VappName}}.name
   org_network_name   = vcd_network_routed_v2.{{.NetworkName}}.name 
+
+  depends_on = [vcd_network_routed_v2.{{.NetworkName}}]
+}
+
+resource "vcd_vapp_org_network" "isolated" {
+  org                = "{{.Org}}"
+  vdc                = "{{.Vdc}}"
+  vapp_name          = vcd_vapp.{{.VappName}}.name
+  org_network_name   = vcd_network_isolated_v2.isolated-test.name 
+
+  depends_on = [vcd_network_isolated_v2.isolated-test]
+}
+
+resource "vcd_vapp_org_network" "imoported" {
+  org                = "{{.Org}}"
+  vdc                = "{{.Vdc}}"
+  vapp_name          = vcd_vapp.{{.VappName}}.name
+  org_network_name   = vcd_nsxt_network_imported.imported-test.name 
+
+  depends_on = [vcd_nsxt_network_imported.imported-test]
```

Comment:
What is the reason for using `depends_on` in these tests when we have reference in `org_network_name`? Is it to avoid possible ordering issues during `destroy`?

Reply:
Yes, as we have seen these problems are possible when not `.id` field is used.